### PR TITLE
Provide meaningful naming for saved image files

### PR DESCRIPTION
**Pull request by engcom-Golf**
### Description (*)
Provided meaningful naming for saved image files
Generated based on title

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#323: Move authentication to AdobeIMS module
2. magento/adobe-stock-integration#328: User sets the image name

### Manual testing scenarios (*)

1. Save preview image
2. Verify image and image name.

### CODE REVIEW

File: AdobeStockImage/Model/Storage.php [hunk 1]
```diff
@@ -101,6 +103,19 @@ public function save(string $imageUrl, string $destinationDirectoryPath = '') :
...
 
+    /**
+     * Generate image name by Title + id.
+     *
+     * @param aray $imageData
+     * @return string
+     */
+    private function generateImageName(array $imageData) :string
+    {
+        $imgName = str_replace(' ', '', $imageData['title']).$imageData['id'];
```
Comment (sivaschenko) [inline AdobeStockImage/Model/Storage.php]: let's replace spaces in the original name with dashes ('-') and shorten the file name to i.e. 32 characters (as titles can be really long)

File: AdobeStockImage/Model/Storage.php [hunk 2]
```diff
@@ -71,24 +72,25 @@ public function __construct(
...
      * @throws LocalizedException
      */
-    public function save(string $imageUrl, string $destinationDirectoryPath = '') : string
+    public function save(Asset $assets, string $destinationDirectoryPath = '') : string
     {
         if (!empty($destinationDirectoryPath)) {
             $destinationDirectoryPath = rtrim($destinationDirectoryPath, '/') . '/';
         }
-        $destinationPath = $destinationDirectoryPath . $this->getFileName($imageUrl);
+        $imgName = $this->generateImageName($assets->getData());
```
Comment (sivaschenko) [inline AdobeStockImage/Model/Storage.php]: Please avoid unnecessary variables shortening, better readability is a priority in comparison to saving 2 chars
```suggestion
        $imageName = $this->generateImageName($assets->getData());
```

File: AdobeStockImage/Model/Storage.php (hunk 1 above)
Comment (sivaschenko) [inline AdobeStockImage/Model/Storage.php]:
```suggestion
        $imageName = str_replace(' ', '', $imageData['title']).$imageData['id'];
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @engcom-Golf @Nazar65 ! Can you please add more testcases (ideally with dataprovider) to unit test to test different image titles (ideally let's take a couple of real image names and test some edge cases, special chars etc.)

File: AdobeStockImage/Model/Storage.php (hunk 1 above)
Comment (engcom-Golf) [reply AdobeStockImage/Model/Storage.php]: ok

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @engcom-Golf thanks for the PR. Can you please update the implementation to allow user to set custom name as well.

Here is a story for this functionality https://github.com/magento/adobe-stock-integration/issues/328

Also, please correct the PR description

**Comment by engcom-Golf** [comment]
HI @sivaschenko could you please review new changes ?

**Review by sivaschenko — DISMISSED** [review]

**Review by sivaschenko — DISMISSED** [review]

**Review by sivaschenko — APPROVED** [review]
